src/frontend/AppContainerWrapper.tsx

```diff
@@ -94,11 +98,44 @@ function hidePracticeModeTemporarily(route: string | null) {
 
 const AppContainerWrapper = () => {
   const navRef = React.useRef<NavigationContainerComponent>();
+  const appState = React.useRef(AppState.currentState);
   const [inviteModalEnabled, setInviteModalEnabled] = React.useState(true);
   const [queuedInvite, setQueuedInvite] = React.useState<string | null>(null);
   const [{ experiments }] = React.useContext(SettingsContext);
   const [hidePracticeBar, setHidePracticeBar] = React.useState(true);
   const [hidePracticeMode, setHidePracticeMode] = React.useState(false);
+  const [authState, setAppState] = React.useContext(SecurityContext);
+
+  React.useEffect(() => {
+    const appStateListener = AppState.addEventListener(
+      "change",
+      handleStateAndSetCheckPassFlag
+    );
+
+    return () => appStateListener.remove();
+  }, [authState.authStatus]);
+
+  const handleStateAndSetCheckPassFlag = (nextAppState: AppStateStatus) => {
+    console.log("newAppStatus " + nextAppState);
+    console.log("appState");
+    if (
+      appState.current.match(/active/) &&
+      nextAppState === "background" &&
+      authState.authStatus !== "notRequired"
+    ) {
+      setAppState({ type: "setAuthStatus", newAuthStatus: "pending" });
+    }
+  };
+
+  const AppContainer = React.useMemo(() => {
+    if (authState.authStatus === "pending") {
+      return AuthContainer;
+    }
+
+    return experiments.onboarding ? OnboardingContainer : DefaultContainer;
+  }, [experiments.onboarding, authState.authStatus]);
+
```

Comment:
Our navigation stacks are becoming a little confusing, so I followed @achou11's lead here and took advantage of the functional component useMemo Hook save the current appContainer